### PR TITLE
Extend `NONMINIMAL_BOOL` lint

### LINKED ISSUE #5794: Lint double negation `!count != 0`
### What it does

I think, clippy should lint the following code:
```rust
if !count != 0 {
}
```

My brain melts trying to read this expression. It's not _that_ difficult to read, just something immediately throwing me off.

### Drawbacks

None.

### Example

```rust
if !count != 0 {
}
```

Could be written as:

```rust
if count == 0 {
}
```

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #5794.

r? @blyxyas 

changelog: Extend `NONMINIMAL_BOOL` lint

### CODE REVIEW

**Review by blyxyas — APPROVED** [review]
This could be merged, but to fix #5794 this would need a test for `!a != 12`

**Comment by GuillaumeGomez** [comment]
Noted. Gonna add support for this case.

**Comment by GuillaumeGomez** [comment]
Euh wait, no. It's not the same, unless `a` is a bool type. I'll add a check only for this case.

**Comment by Centri3** [comment]
Shouldn't this be true for integers as well? e.g., `!a != 12` where `a` is `12` is true, as `Not` flips each bit of the integer

**Comment by Centri3** [comment]
Oh wow nvm, I thought this was for bool only. It already lints integers too

**Comment by GuillaumeGomez** [comment]
Anything else to be done here then? :)

File: clippy_lints/src/booleans.rs [hunk 1]
```diff
@@ -85,7 +85,111 @@ impl<'tcx> LateLintPass<'tcx> for NonminimalBool {
     ) {
         NonminimalBoolVisitor { cx }.visit_body(body);
     }
+
+    fn check_expr(&mut self, cx: &LateContext<'tcx>, expr: &'tcx Expr<'tcx>) {
+        match expr.kind {
+            ExprKind::Unary(UnOp::Not, sub) => check_inverted_condition(cx, expr.span, sub),
+            ExprKind::Binary(op, left, right) if matches!(op.node, BinOpKind::Eq | BinOpKind::Ne) => {
```
Comment (blyxyas) [inline clippy_lints/src/booleans.rs:92]: this should have a comment talking about what case they cover, just for clarity

**Review by blyxyas — APPROVED** [review]
Just a mini-nit, could you squash the commits?

File: clippy_lints/src/booleans.rs (hunk 1 above)
Comment (GuillaumeGomez) [reply clippy_lints/src/booleans.rs]: Sure done.

**Comment by GuillaumeGomez** [comment]
> Just a mini-nit, could you squash the commits?

I squashed the commits in two commits as I still want to keep the "inverted boolean in comparison" in its own commit as it feels like a different kind of check.

**Comment by RalfJung** [comment]
I think this probably introduced https://github.com/rust-lang/rust-clippy/issues/12371

**Pull request merged**